### PR TITLE
Update docs for Module.

### PR DESCRIPTION
**Pull request by connorshea**
And add some missing method parameters to Module methods. A lot of methods have an `inherit` parameter that wasn't in any of the signatures.

### CODE REVIEW

File: stdlib/builtin/module.rbs [hunk 1]
```diff
@@ -1,248 +1,1069 @@
...
+  # If a method has protected visibility, it is callable only where `self` of the
+  # context is the same as the method. (method definition or instance_eval). This
+  # behavior is different from Java's protected method. Usually `private` should
+  # be used.
+  #
+  # Note that a protected method is slow because it can't use inline cache.
+  #
+  # To show a private method on RDoc, use `:doc:` instead of this.
+  #
   def protected: (*Symbol | String arg0) -> self
```
Comment (connorshea) [inline stdlib/builtin/module.rbs]: Should this use backticks (\`) since protected is a keyword?

File: stdlib/builtin/module.rbs [hunk 2]
```diff
@@ -1,248 +1,1069 @@
...
+  #
+  def singleton_class?: () -> bool
+
+  # Returns a string representing this module or class. For basic classes and
+  # modules, this is the name. For singletons, we show information on the thing
+  # we're attached to as well.
+  #
   def to_s: () -> String
 
   def undefMethod: (Symbol | String arg0) -> self
```
Comment (connorshea) [inline stdlib/builtin/module.rbs]: I asked in the Sorbet Slack about this, but I think they accidentally typo'd this and it should be `undef_method`?

Comment (soutaro) [reply stdlib/builtin/module.rbs:1055]: Good find! It should be `undef_method`. I think you can push another commit to fix it in this PR.

**Comment by connorshea** [comment]
@soutaro fixed 👍

**Pull request merged**